src/predicate_lang/predicate_lang.mli

```diff
@@ -30,36 +30,38 @@ val encode : 'a Encoder.t -> 'a t Encoder.t
 
 val to_dyn : 'a Dyn.builder -> 'a t Dyn.builder
 
-val exec : 'a t -> standard:'a t -> ('a -> bool) -> bool
+val exec : 'a t -> standard:'a t -> equal:('a -> 'a -> bool) -> 'a -> bool
 
 val empty : 'a t
 
-val map : 'a t -> f:('a -> 'b) -> 'b t
-
-val to_predicate :
-  'a Predicate.t t -> standard:'a Predicate.t t -> 'a Predicate.t
+val compare : ('a -> 'a -> Ordering.t) -> 'a t -> 'a t -> Ordering.t
 
 module Glob : sig
-  (** The underlying type for the string predicate is not an actual glob, so
-      this module is confusingly named. *)
-
-  type glob
+  module Element : sig
+    type t =
+      | Glob of Dune_glob.V1.t
+      | Literal of string
+  end
 
-  type nonrec t = glob t
+  type nonrec t = Element.t t
 
   val to_dyn : t -> Dyn.t
 
   val exec : t -> standard:t -> string -> bool
 
   val filter : t -> standard:t -> string list -> string list
 
-  val create_glob : Dune_glob.V1.t -> glob
-
   val of_glob : Dune_glob.V1.t -> t
 
-  val of_pred : (string -> bool) -> t
+  val of_string_list : string list -> t
```

Comment:
What does this match? Any of the strings in the list? It would be nice to add some comments.